bot/cogs/moderation/infractions.py

```diff
@@ -494,14 +494,17 @@ def mod_log(self) -> ModLog:
             f"{dm_result}{confirm_msg} **{infr_type}** to {user.mention}{expiry_msg}{end_msg}."
         )
 
+        # Check if `avatar_url_as` is a method, if it isn't we know it's a proxy object.
+        user_plain = f" (`{user}`)" if hasattr(user.avatar_url_as, "__self__") else ""
```

Comment:
While this is functional, it's really opaque and pretty fragile.

It would be a better idea to go with what you suggested in #dev-contrib this morning and not implement our proxy user essentially identically in two different places ([1](https://github.com/python-discord/bot/blob/master/bot/cogs/moderation/utils.py#L33-L48), [2](https://github.com/python-discord/bot/blob/master/bot/cogs/watchchannels/watchchannel.py#L27-L40)) and instead create a proper object that can be shared & utilized by `isinstance` rather than relying on an attribute check.

Reply:
Where should the new object be defined?
In one of the places where it's now, or in `bot/utils` as a new module?